routers/web/repo/compare.go

```diff
@@ -947,6 +949,41 @@ func ExcerptBlob(ctx *context.Context) {
 			section.Lines = append(section.Lines, lineSection)
 		}
 	}
+	if ctx.Doer != nil {
+		ctx.Data["SignedUserID"] = ctx.Doer.ID
+	}
+	isPull := ctx.FormBool("is_pull")
+	ctx.Data["PageIsPullFiles"] = isPull
+
+	if isPull {
+		issueIndex := ctx.FormInt64("issue_index")
```

Comment:
We need to check whether the user have read permission of this issue/pull unit of this repository.

Reply:
Added permission check for PR